Description: Lower CPU usage by avoiding pointless copy.
 Upstream's commit message:
 Avoid pointlessly doing a full copy of a linked list every frame when the
 copy is never changed and never different to the source. This cuts an
 instant 6-9% from compiz' CPU usage (LP: #946887)

=== modified file 'plugins/opengl/src/paint.cpp'
--- a/plugins/opengl/src/paint.cpp	2012-02-27 03:53:45 +0000
+++ b/plugins/opengl/src/paint.cpp	2012-03-05 09:38:48 +0000
@@ -217,9 +217,6 @@
     GLMatrix      vTransform;
     CompPoint     offXY;
 
-    CompWindowList                   pl;
-    CompWindowList::reverse_iterator rit;
-
     unredirectFS = CompositeScreen::get (screen)->
 	getOption ("unredirect_fullscreen_windows")->value ().b ();
 
@@ -234,7 +231,8 @@
 	count	       = 0;
     }
 
-    pl = cScreen->getWindowPaintList ();
+    const CompWindowList &pl = cScreen->getWindowPaintList ();
+    CompWindowList::const_reverse_iterator rit;
 
     if (!(mask & PAINT_SCREEN_NO_OCCLUSION_DETECTION_MASK))
     {

